crates/pixi_utils/src/prefix.rs

```diff
@@ -92,7 +95,18 @@ impl Prefix {
                         })
                     })
             })
-            .collect();
+            .collect::<Vec<_>>();
+
+        // Also scan binary folders for symlinks that aren't tracked in prefix records
+        if let Ok(additional_executables) = self.scan_binary_folders_for_executables() {
```

Comment:
I worry about is the possible performance regression, did you test it on like a heavy environment?

I stress test with `ros-jazzy-desktop` environments.